packages/delisp-core/src/convert.ts

```diff
@@ -137,40 +137,17 @@ defineConversion("if", expr => {
 });
 
 function parseLetBindings(bindings: ASExpr): SLetBinding[] {
-  if (bindings.type !== "list") {
+  if (bindings.type !== "map") {
     throw new Error(
-      printHighlightedExpr(`'let' bindings should be a list`, bindings.location)
+      printHighlightedExpr(`'let' bindings should be a map`, bindings.location)
     );
```

Comment:
When happens if we define a binding for a keyword like `:x`?

Do local variables take precedence over inline primitives? So..

```
(let {:x 10}
  x)
```

returns 10?  I don't think so, but it should. Or we should throw an error in convert.

Reply:
It's very important that the scope rules of the compiler & the type inference match, actually. I'll add a ticket to refactor a bit to make those bugs less likely.